### ISSUE TITLE
Unicode escapes in `#inspect` and `#dump`

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#11405]
Unicode escape sequences produced by `#inspect` and `#dump` are different for `Char` and `String`.
The latter uses the fixed four digit format for codepoints `U+0000` through `U+FFFF`. `Char` on the other hand, uses the variable length format wrapped in curly braces. `String` uses uppercase letters, `Char` lowercase.

```cr
"\u00C1".inspect # => %("\\u00C1")
'\u{C1}'.inspect # => %('\\u{c1}')
"\u001C".dump    # => %("\\u001C")
'\u{1c}'.dump    # => %('\\u{1c}')
```

This difference for exactly the same behaviour is surprising and unsubstantiated. For concistency, we should adapt a unique format style for Unicode escape sequences.

I'm convinced the four-digit format is preferable. A constant width for escape sequences creates less visual noise than variable width with curly braces. The four-digit escape sequences have the equal or shorter length, except for the first 15 codepoints (and half of them have dedicated single-char escapes).
Hexadecimal formats for Unicode codepoints commonly use upper case letters.

To sum up, the behaviour of `String` is best and should be adapted for `Char`.

I'm not sure if this is to be considered a breaking change. The semantics are identical, just the format changes. So I'd suggest classifying it as non-breaking.
